zim/gui/pageview/__init__.py

```diff
@@ -515,10 +515,44 @@ def image_file_load_pixels(file, width_override=-1, height_override=-1):
 	except:
 		logger.debug('GTK failed to read image, using Pillow fallback: %s', file.path)
 
+		import PIL
 		from PIL import Image # load Pillow only if necessary
 
 		with Image.open(file.path) as img_pil:
+			def rotate_pillow_image_by_exif_orientation(image: Image.Image) -> Image.Image:
+				tag_id_orientation = 0x0112  # noqa: WPS339
+				# https://pillow.readthedocs.io/en/stable/releasenotes/6.0.0.html#added-exif-class
+				pillow_version = tuple(map(int, PIL.__version__.split('.')))
+				if pillow_version >= (6, 0, 0):
```

Comment:
Good reference linking! Debian [old-stable still has 5.4.1](https://tracker.debian.org/pkg/pillow) (last updated on 2022-02-05) on [Python 3.7.3](https://tracker.debian.org/pkg/python3-defaults), so support is appreciated.